### PR TITLE
Enable JSPI testing for node version 19.

### PR DESCRIPTION
**Pull request by brendandahl**

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -571,6 +571,28 @@ def require_wasm_eh(self):
...
+        self.node_args += exp_args
+        return
+
+    if config.V8_ENGINE and config.V8_ENGINE in self.js_engines:
+      self.emcc_args.append('-sENVIRONMENT=shell')
+      self.js_engines = [config.V8_ENGINE]
+      self.v8_args += exp_args
+      return
+
+    if 'EMTEST_SKIP_EH' in os.environ:
```
Comment (sbc100) [inline test/common.py:591]: You would want to change this to `EMTEST_SKIP_JSPI` here and on the lines below.

File: test/test_core.py [hunk 2]
```diff
@@ -230,10 +230,7 @@ def with_asyncify_and_stack_switching(f):
   def metafunc(self, stack_switching):
     if stack_switching:
       self.set_setting('ASYNCIFY', 2)
-      self.require_v8()
-      # enable stack switching and other relevant features (like reference types
-      # for the return value of externref)
-      self.v8_args.append('--experimental-wasm-stack-switching')
+      self.require_wasm_jspi()
```
Comment (sbc100) [inline test/test_core.py:233]: I think this can just be called `require_jspi`.

**Review by sbc100 — APPROVED** [review]

**Comment by hoodmane** [comment]
In my experience the version of JSPI shipped with node 19 is broken -- if you try to use it, it segfaults sometimes.

**Comment by hoodmane** [comment]
In my experience the version of JSPI shipped with node 19 is broken -- if you try to use it, it segfaults sometimes.

**Comment by brendandahl** [comment]
Good to know. For CI we'll only be testing on canary which is currently at 20, so this hopefully shouldn't be an issue.

**Pull request merged**